src/sentry/models/user.py

```diff
@@ -185,14 +185,24 @@ def delete(self):
             avatar = self.avatar.first()
             if avatar:
                 avatar.delete()
-            for outbox in User.outboxes_for_update(self.id):
+            for outbox in self.outboxes_for_update():
                 outbox.save()
             return super().delete()
 
+    def update(self, *args, **kwds):
+        with transaction.atomic(), in_test_psql_role_override("postgres"):
+            for outbox in self.outboxes_for_update():
+                outbox.save()
```

Comment:
Since this pattern is being repeated in numerous places, is there worth in adding a `queue_outboxes_for_update` method that performs this loop + persistence?

Reply:
I see the concern given its repetition, but I also don't like the idea yet of encapsulating these two lines like that.  I, prefer?  Keeping the logic exposed because it honestly makes really clear the composed pieces.